modules/node_modules/@frogpond/ccci-stolaf-college/v1/jobs/index.js

```diff
@@ -1,52 +1,85 @@
 import {get, ONE_DAY} from '@frogpond/ccc-lib'
 import mem from 'mem'
+import pMap from 'p-map'
 import _jsdom from 'jsdom'
 import getUrls from 'get-urls'
 
 const {JSDOM} = _jsdom
 
-export function cleanJob(job) {
-	// these all need to retain their newlines
-	const description = cleanTextBlock(
-		JSDOM.fragment(job.description).textContent,
-	)
-	const comments = cleanTextBlock(JSDOM.fragment(job.comments).textContent)
-	const skills = cleanTextBlock(JSDOM.fragment(job.skills).textContent)
-	const howToApply = cleanTextBlock(JSDOM.fragment(job.howToApply).textContent)
-	const timeline = cleanTextBlock(JSDOM.fragment(job.timeline).textContent)
-	const timeOfHours = cleanTextBlock(
-		JSDOM.fragment(job.timeOfHours).textContent,
-	)
-
-	const contactEmail = fixupEmailFormat(job.contactEmail)
-	const contactPhone = fixupPhoneFormat(job.contactPhone)
+const jobsUrl =
+	'https://wp.stolaf.edu/student-jobs/wp-json/wp/v2/pages/80'
+
+const PARAGRAPHICAL_KEYS = ['Job Description', 'Skills Needed', 'Additional Comments', 'How to Apply', 'Hiring Timeline']
+
+async function fetchDetail(url) {
+	const resp = await get(url)
+
+	// run-scripts value is needed to properly evaluate javascript to display an email address.
+	// see the jsdom documentation for more details https://github.com/jsdom/jsdom#executing-scripts
+	const dom = new JSDOM(resp.body, {contentType: 'text/html', runScripts: 'dangerously' })
+
+	const id = url.replace(/\D/g,'')
+	const titleText = dom.window.document.querySelector('.gv-list-view-title > h3').textContent
+	const details = dom.window.document.querySelectorAll('div')
+
+	const detailMap = [...details].reduce((coll, listEl) => {
+		let [key, ...value] = listEl.childNodes
```

Comment:
Here `key` is a single node, but `value` is either an `Array` or a `NodeList` — not entirely sure. (a) is this even correct? (b) can `value` be given a different name so that we can tell what it would be expected to hold?

Reply:
I can look into extracting all of this as a testable function to help us understand how this behaves.

Reply:
`value` can be one of many things, depending upon the page structure, yeah. I'll go ahead and rename it so that it reflects its relation to the dom.